services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/serialization/CrtUploadResumeToken.java

```diff
@@ -0,0 +1,101 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.transfer.s3.internal.serialization;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Objects;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.protocols.jsoncore.JsonNode;
+import software.amazon.awssdk.protocols.jsoncore.JsonNodeParser;
+import software.amazon.awssdk.protocols.jsoncore.JsonWriter;
+
+@SdkInternalApi
+public final class CrtUploadResumeToken {
+
+    private final Long totalNumOfParts;
+    private final Long partSizeInBytes;
+    private final String multipartUploadId;
+
+    public CrtUploadResumeToken(Long totalNumOfParts, Long partSizeInBytes, String multipartUploadId) {
+        this.totalNumOfParts = totalNumOfParts;
+        this.partSizeInBytes = partSizeInBytes;
+        this.multipartUploadId = multipartUploadId;
+    }
+
+    public Long totalNumOfParts() {
+        return totalNumOfParts;
+    }
+
+    public Long partSizeInBytes() {
+        return partSizeInBytes;
+    }
+
+    public String multipartUploadId() {
+        return multipartUploadId;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        CrtUploadResumeToken token = (CrtUploadResumeToken) o;
+
+        if (!Objects.equals(totalNumOfParts, token.totalNumOfParts)) {
+            return false;
+        }
+        if (!Objects.equals(partSizeInBytes, token.partSizeInBytes)) {
+            return false;
+        }
+        return Objects.equals(multipartUploadId, token.multipartUploadId);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = totalNumOfParts != null ? totalNumOfParts.hashCode() : 0;
+        result = 31 * result + (partSizeInBytes != null ? partSizeInBytes.hashCode() : 0);
+        result = 31 * result + (multipartUploadId != null ? multipartUploadId.hashCode() : 0);
+        return result;
+    }
+
+    public static String marshallResumeToken(CrtUploadResumeToken resumeToken) {
+        JsonWriter jsonGenerator = JsonWriter.create();
+        jsonGenerator.writeStartObject();
+
+        TransferManagerJsonMarshaller.LONG.marshall(resumeToken.totalNumOfParts(), jsonGenerator, "total_num_parts");
+        TransferManagerJsonMarshaller.LONG.marshall(resumeToken.partSizeInBytes(), jsonGenerator, "partition_size");
+        TransferManagerJsonMarshaller.STRING.marshall("AWS_S3_META_REQUEST_TYPE_PUT_OBJECT", jsonGenerator, "type");
+        TransferManagerJsonMarshaller.STRING.marshall(resumeToken.multipartUploadId(), jsonGenerator, "multipart_upload_id");
+
+        jsonGenerator.writeEndObject();
+        return new String(jsonGenerator.getBytes(), StandardCharsets.UTF_8);
+    }
```

Comment:
Should this be a virtual method instead of a static one? (`String x = resumeToken.marshall()`)